widget/accordioncontainer.go

```diff
@@ -0,0 +1,335 @@
+	width := 0
+	height := 0
+	for _, i := range r.container.items {
+		min := i.MinSize()
+		width = fyne.Max(width, min.Width)
+		height += min.Height
+	}
+	if len(r.container.items) > 0 {
+		width += theme.Padding() * 2
+		height += theme.Padding() * 2
+	}
+	return fyne.NewSize(width, height)
+}
+
+func (r *accordionContainerRenderer) Layout(size fyne.Size) {
+	x := theme.Padding()
+	y := theme.Padding()
+	w := size.Width - theme.Padding()*2
+	for _, i := range r.container.items {
+		h := i.MinSize().Height
+		i.Move(fyne.NewPos(x, y))
+		i.Resize(fyne.NewSize(w, h))
+		y += h
+	}
+}
+
+func (r *accordionContainerRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (r *accordionContainerRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *accordionContainerRenderer) Refresh() {
+	for _, i := range r.container.items {
+		i.Refresh()
+	}
+	canvas.Refresh(r.container)
+}
+
+func (r *accordionContainerRenderer) Destroy() {
+}
+
+var _ fyne.Widget = (*accordionItem)(nil)
+
+type accordionItem struct {
+	BaseWidget
+	container *AccordionContainer
+	header    *accordionItemHeader
+	detail    fyne.CanvasObject
+	open      bool
+}
+
+func (i *accordionItem) setOpen(open bool) {
+	if i.open == open {
+		return
+	}
+	if i.open = open; i.open {
+		i.detail.Show()
+	} else {
+		i.detail.Hide()
+	}
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (i *accordionItem) CreateRenderer() fyne.WidgetRenderer {
+	return &accordionItemRenderer{
+		item: i,
+		objects: []fyne.CanvasObject{
+			i.header,
+			i.detail,
+		},
+	}
+}
+
+type accordionItemRenderer struct {
+	item    *accordionItem
+	objects []fyne.CanvasObject
+}
+
+func (r *accordionItemRenderer) MinSize() fyne.Size {
+	min := r.item.header.MinSize()
+	width := min.Width
+	height := min.Height
+	if r.item.open {
+		min := r.item.detail.MinSize()
+		width = fyne.Max(width, min.Width)
+		height += min.Height
+	}
+	return fyne.NewSize(width, height)
+}
+
+func (r *accordionItemRenderer) Layout(size fyne.Size) {
+	height := r.item.header.MinSize().Height
+	r.item.header.Move(fyne.NewPos(0, 0))
+	r.item.header.Resize(fyne.NewSize(size.Width, height))
+	if r.item.open {
+		r.item.detail.Move(fyne.NewPos(0, height))
+		r.item.detail.Resize(fyne.NewSize(size.Width, r.item.detail.MinSize().Height))
+	}
+}
+
+func (r *accordionItemRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (r *accordionItemRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *accordionItemRenderer) Refresh() {
+	r.item.header.Refresh()
+	r.item.detail.Refresh()
+	canvas.Refresh(r.item)
+}
+
+func (r *accordionItemRenderer) Destroy() {
+}
+
+var _ fyne.Tappable = (*accordionItemHeader)(nil)
+var _ fyne.Widget = (*accordionItemHeader)(nil)
+var _ desktop.Hoverable = (*accordionItemHeader)(nil)
+
+type accordionItemHeader struct {
```

Comment:
This seems like a re-implementation of button, did you consider using button itself?

Reply:
Switched to use buttons for header, what do you think of the look&feel?

Reply:
Looking at this with fresh eyes, button provides the required functionality and makes it more obvious that the header can be tapped. However the centered layout is not ideal, I'd prefer the text to be leading-aligned with the label either all the way left, or all the way right:
```
+---------------------------------+
| Icon Header                     |
+---------------------------------+
```
or
```
+---------------------------------+
| Header                     Icon |
+---------------------------------+
```
Unfortunately button does not provide a lot of flexibility here.

@pbarnum what do you think?

Reply:
The second diagram works better than the first, mainly because if some accordions don't have icons, the headers will still be aligned.

Reply:
I just realized that the icon would most likely be the open/close icons... In which case I still think the second option is the best.

Reply:
I went with
```
+---------------------------------+
| Icon Header                     |
+---------------------------------+
```
to avoid being too similar to Select, but we can revisit this in future